src/m365/onenote/commands/page/page-list.ts

```diff
@@ -0,0 +1,155 @@
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupName: typeof args.options.groupName !== 'undefined',
+        webUrl: typeof args.options.webUrl !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '--userId [userId]' },
+      { option: '--userName [userName]' },
+      { option: '--groupId [groupId]' },
+      { option: '--groupName [groupName]' },
+      { option: '-u, --webUrl [webUrl]' }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.webUrl) {
+          return validation.isValidSharePointUrl(args.options.webUrl);
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({
+      options: ['userId', 'userName', 'groupId', 'groupName', 'webUrl'],
+      runsWhen: (args) => {
+        const options = [args.options.userId, args.options.userName, args.options.groupId, args.options.groupName, args.options.webUrl];
+        return options.some(item => item !== undefined);
+      }
+    });
+  }
+
+  private async getEndpointUrl(args: CommandArgs): Promise<string> {
+    let endpoint: string = `${this.resource}/v1.0/`;
+
+    if (args.options.userId) {
+      endpoint += `users/${args.options.userId}`;
+    }
+    else if (args.options.userName) {
+      endpoint += `users/${args.options.userName}`;
+    }
+    else if (args.options.groupId) {
+      endpoint += `groups/${args.options.groupId}`;
+    }
+    else if (args.options.groupName) {
+      const groupId = await this.getGroupId(args.options.groupName);
+      endpoint += `groups/${groupId}`;
+    }
+    else if (args.options.webUrl) {
+      const siteId = await this.getSpoSiteId(args.options.webUrl);
+      endpoint += `sites/${siteId}`;
+    }
+    else {
+      endpoint += 'me';
+    }
+    endpoint += '/onenote/pages';
+    return endpoint;
+  }
+
+  private async getGroupId(groupName: string): Promise<string> {
+    const group = await aadGroup.getGroupByDisplayName(groupName);
+    return group.id!;
+  }
+
+  private async getSpoSiteId(webUrl: string): Promise<string> {
+    const url = new URL(webUrl);
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/sites/${url.hostname}:${url.pathname}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const site = await request.get<{ id: string }>(requestOptions);
+    return site.id;
+  }
```

Comment:
I wonder wouldn't it make more sense to reuse `m365 spo site get` instead of doing the same reuqest here 🤔.
what do you think?
@pnp/cli-for-microsoft-365-maintainers any comments?

Reply:
We've already said that we would go and create a util for this, as specified in #4316, however, in this util, we might use this command for sure.

Reply:
not sure of the approach where a command would become part of an util 🤔. But for sure I would rather reuse existing implementation if we already somewhere else do the same kind of request 👍

Reply:
If we create a util to get the ID of something, I would write the request with a `select` query rather than reusing a command that fetches unneeded data.

Reply:
> reuqest
Also, we are retrieving the Graph ID of the site here. If we use spo site get, we won't retrieve the MS Graph ID. I've also checked the output of the command, and it isn't available in the response object.

Reply:
ok clear